stepflow-rs/crates/stepflow-cli/src/cli.rs

```diff
@@ -480,180 +436,199 @@ impl Cli {
                 }
 
                 let executor = WorkflowLoader::create_executor_from_config(config).await?;
-                let input = input_args.parse_input(true)?;
-
                 let flow_id =
                     BlobId::from_flow(&flow).change_context(crate::MainError::Configuration)?;
-                let (run_id, output) = run(executor, flow, flow_id, input, overrides).await?;
-                // Output run_id without hyphens for Jaeger trace ID compatibility
-                eprintln!("run_id: {}", run_id.simple());
-                output_args.write_output(output)?;
-            }
-            #[allow(clippy::print_stdout)]
-            Command::RunBatch {
-                flow_path,
-                inputs_path,
-                max_concurrent,
-                output_path,
-                config_args,
-                execution_args,
-            } => {
-                use stepflow_plugin::Context as _;
-
-                let flow: Arc<Flow> = load(&flow_path)?;
-                let flow_dir = flow_path.parent();
-                let config = config_args.load_config(flow_dir)?;
 
-                // Validate workflow and configuration before execution
-                let diagnostics = stepflow_analysis::validate_with_config(
-                    &flow,
-                    &config.plugins,
-                    &config.routing,
-                )
-                .change_context(crate::MainError::ValidationError(
-                    "Validation failed".to_string(),
-                ))?;
+                // Check if we're in batch mode
```

Comment:
Ideally we can unify this more -- we can make the JSONL loading part of the InputARgs so we can just do ```inputs.load_inputs()``` and then we can determine how to write results based on an `inputs.is_batch()` or some such.